2023/incoming/perl-the-humane-programming-language.pod

```diff
@@ -0,0 +1,90 @@
+Author: perl@dnmfarrell.com
+Title: Perl - The Humane Programming Language
+Topic: 
+
+state of the onion first three talks
+people are different
+
+=encoding utf8
+
+=head2 Perl - The Humane Programming Language
+
+Human languages follow the L<Law of Brevity|https://en.wikipedia.org/wiki/Brevity_law>, where the most commonly used words are also the shortest. Perl aims for brevity too. Its keywords tend to be shorter than the traditional versions:
+
+=begin code
+
+Traditional    Perl
+-----------    ----
+break          last
+continue       next
+filter         grep
+function       sub
+import         use
+let            my
+throw          die
+
+=end code
+
+Often entire words can be omitted. Subroutine declarations do not require a C<return> statement, and may be called without parens. Built-in functions can operate on default variables like C<$_> instead of an explicit arg. This code loops over each line of input and splits the tab-separated input into an array, without ever referencing the input variable:
+
+=begin perl
+
+while (<>) {
+  chomp;
+  my @cols = split /\t/;
+  ...
+}
+
+=end perl
+
+> People like things to be visually distinct from their surroundings. That's also why the various classes of operators and variables in Perl are visually distinct from each other. It's just sound human engineering.
+> Larry Wall
+> 
+
+By making variables, operators and function names visually distinct, Perl saves you reading time too.
+
+=head3 Do What I Mean
+
+Perl might be the most dynamic of the dynamic languages. Take this pseudocode:
+
+=begin code
+
+foo = "100"
+bar = foo + 1
+
+=end code
+
+Python and Ruby error on code like this. Since they resolve C<foo> to a string type, they refuse to concatenate it with the number 1 later on.
+
```

Comment:
constant, annoying coercions of object to strings when the language already knows the operator wants a string. Larry has said that Perl is a language of verbs (operators) instead of nouns. The verbs decide how to think about the data.